### PR TITLE
use shared libraries on FreeBSD

### PR DESCRIPTION
**Pull request by MartinNowak**

### CODE REVIEW

**Comment by MartinNowak** [comment]
updated

**Comment by MartinNowak** [comment]
Fixed the remaining issues https://github.com/D-Programming-Language/dmd/pull/4838, https://github.com/D-Programming-Language/druntime/pull/1333.

**Comment by MartinNowak** [comment]
This might need an auto-tester update, b/c the shared phobos library is picked by default but the LD_LIBRARY_PATH isn't set accordingly.

**Comment by MartinNowak** [comment]
requires https://github.com/D-Programming-Language/dmd/pull/4975

**Comment by MartinNowak** [comment]
This is just about building and testing, not making them the default.

File: posix.mak [hunk 1]
```diff
@@ -266,12 +266,15 @@ OBJS = $(addsuffix $(DOTOBJ),$(addprefix $(ROOT)/,$(C_MODULES)))
 
 MAKEFILE = $(firstword $(MAKEFILE_LIST))
 
+# build with shared library support
```
Comment (andralex) [inline posix.mak]: Incomplete comment.

     # Build with shared library support on Linux and FreeBSD.

File: posix.mak [hunk 2]
```diff
@@ -266,12 +266,15 @@ OBJS = $(addsuffix $(DOTOBJ),$(addprefix $(ROOT)/,$(C_MODULES)))
...
+SHARED=$(if $(findstring $(OS),linux freebsd),1,)
+
 ################################################################################
 # Rules begin here
 ################################################################################
 
 # Main target (builds the dll on linux, too)
-ifeq (linux,$(OS))
+ifneq (,$(SHARED))
```
Comment (andralex) [inline posix.mak]: I'd use `ifeq (1,$(SHARED))` but I guess there's no need to be super cautious.

**Comment by MartinNowak** [comment]
Updated, still blocked by the pending dmd PR though.

**Comment by MartinNowak** [comment]
Updated, should work now.

**Pull request merged**